app/javascript/components/forms/DeleteUserForm.jsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import {
+  Button,
+} from 'react-bootstrap';
+import Form from './Form';
+import useDeleteUser from '../../hooks/mutations/users/useDeleteUser';
+import { useAuth } from '../../contexts/auth/AuthProvider';
+
+export default function DeleteUserForm() {
+  const currentUser = useAuth();
+  const methods = useForm();
+  const { onSubmit } = useDeleteUser(currentUser?.id);
+
+  return (
+    <Form methods={methods} onSubmit={onSubmit}>
+      <Button variant="danger" type="submit">
```

Comment:
Since you are using the `mutateAsync` you can leverage the `isSubmitting` react hook form state to disable the delete button avoiding flooding the server and show maybe proper animations like a spinner or anything relevant to that context.